src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        shoppingCartDao.create(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().remove(product);
+        shoppingCartDao.update(shoppingCart);
+        return true;
```

Comment:
```suggestion
        if (shoppingCart.getProducts().remove(product)) {
             shoppingCartDao.update(shoppingCart);
             return true;
        }
        return false;
```
this way it would be more informative